src/gainmap.c

```diff
@@ -382,11 +441,59 @@ avifResult avifFindMinMaxWithoutOutliers(const float * gainMapF, int numPixels,
     return AVIF_RESULT_OK;
 }
 
+static const float kEpsilon = 1e-10f;
+
+// Decides which of 'basePrimaries' or 'altPrimaries' should be used for doing gain map math.
+// The other image (base or alternate) will be converted to this color space before computing
+// the ratio between the two images.
+// If a pixel color is outside of the target color space, some of the converted channel values will be negative.
+// This should be avoided, as the negative values must either be clamped or offset before computing the log2()
+// (since log2 only works on > 0 values). But a large offset causes artefacts when partially applying the gain map.
+// Therefore we want to do gain map math in the larger of the two color spaces.
```

Comment:
"largest"? Not sure

Reply:
I believe both are ok but some people consider the -er verison to be more correct when comparing two things.
According to https://www.merriam-webster.com/grammar/can-you-use-a-superlative-of-two it doesn't matter that much.